OnPageCode.js

```diff
@@ -1,219 +1,411 @@
+      return device.androidPhone() || device.iphone();
+    };
+
+    // tablet
+    device.tablet = function() {
+      return device.ipad() || device.androidTablet();
+    };
+
+    device.desktop = function() {
+      return !device.tablet() && !device.mobile();
+    };
+
+    return findMatch(['mobile', 'tablet', 'desktop']);
+  };
+
+  // eslint-disable-next-line camelcase
+  var PM_Network_POC = {
+
+    'timeoutCorrelators': {},
+
+    // IMP: Which namespace to be used, can be identify using,
+    // 1. Go to publisher URL.
+    // 2. Open developer tool and select console tab
+    // 3. Execute below code in the console,
+    //    3.1 window.owpbjs
+    //    If it returns object with some properties, use owpbjs as prebidNamespace
+    //    else, try changing namespace with pbjs (code: window.pbjs)
+    //    3.2 In case of IDHUB only, use code, window.ihowpbjs
+    // 4. Or connect to JS developer
+
+    'prebidNamespace': 'pbjs',
+
+    'publisherId': 1234,
+
+    // N second after auction end, get the stats for given domains
+    'executionDelayInMs': 2000,
+
+    // run the Network analysis only for the mentioned percentage of traffic
+    'testGroupPercentage': 100,
+
+    'cache': {
+      auction: {}
     },
-    {
-      key: "que_st",
-      name: "Queueing and stalled",
-      timeEndKey: "requestStart",
-      timeStartKey: "startTime"
+
+    'statHatParameters': [
+      {
+        key: 'dns',
+        name: 'DNS Lookup',
+        timeEndKey: 'domainLookupEnd',
+        timeStartKey: 'domainLookupStart'
+      },
+      {
+        key: 'tcp',
+        name: 'TCP Connection',
+        timeEndKey: 'connectEnd',
+        timeStartKey: 'connectStart'
+      },
+      {
+        key: 'que_st',
+        name: 'Queueing and stalled',
+        timeEndKey: 'requestStart',
+        timeStartKey: 'startTime'
+      },
+      {
+        key: 'rs_wfs',
+        name: 'Request Sent and Waiting For Server',
+        timeEndKey: 'responseStart',
+        timeStartKey: 'requestStart'
+      },
+      {
+        key: 'cd',
+        name: 'Content Download',
+        timeEndKey: 'responseEnd',
+        timeStartKey: 'responseStart'
+      },
+      {
+        key: 'dur',
+        name: 'Duration',
+        timeEndKey: 'responseEnd',
+        timeStartKey: 'startTime'
+      }
+    ],
+
+    'populateBidderDataForAuction': (bidderObj, requestObj) => {
+      PM_Network_POC.order++;
+      const bidder = PM_Network_POC.bidders.find(bidder => bidder.bidderCode === bidderObj.bidderCode);
+
+      if (!bidder) {
+        PM_Network_POC.bidders.push({
+          bidderCode: bidderObj.bidderCode,
+          searchName: requestObj.url,
+          atLeastOneBidResUsedInAuction: false,
+          bids: [
+            {
+              order: PM_Network_POC.order,
+              reqPayloadCharCount: requestObj.data.length,
+              reqMethod: requestObj.method
+            }
+          ]
+        });
+      } else {
```

Comment:
We should add the statement `atLeastOneBidResUsedInAuction: false,` in the else condition as well. 
As the bidder is found in the `PM_Network_POC.bidders` and the same bidder `atLeastOneBidResUsedInAuction` flag was set to true, then again in the second auction we should reset the `atLeastOneBidResUsedInAuction` flag to false.
@jlquaccia  + @pm-harshad-mane - Correct me If I am wrong.

Reply:
@pm-nitin-shirsat makes sense, i can make a change for this

Reply:
@pm-nitin-shirsat after looking this over once more, I think we should be ok leaving this as is because of line 424 here: https://github.com/pm-harshad-mane/PrebidNetworkMonitoringUsingWorrkerAndELK/blob/38de132478994bd6748c27a8d4b568f49c1a5396/OnPageCode_timeout.js#L424

On auction init, the reset func gets invoked and clears the bidders array to start again with a clean slate